pkg/util/winutil/winctrlhandler.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+// Package winutil contains Windows OS utilities
+package winutil
+
+import "syscall"
+
+var (
+	setConsoleCtrlHandler = k32.NewProc("SetConsoleCtrlHandler")
+)
+
+const (
+	// CtrlCEvent is code for Cntrl+C event
+	CtrlCEvent = 0
+	// CtrlBreakEvent is code for Cntrl+Break event
+	CtrlBreakEvent = 1
+)
+
+func boolToInt(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+// SetConsoleCtrlHandler sets the handler function for console control events.
+func SetConsoleCtrlHandler(handler func(uint32) bool, add bool) error {
+	ret, _, err := setConsoleCtrlHandler.Call(
+		uintptr(syscall.NewCallbackCDecl(func(sig uint32) uintptr {
```

Comment:
why did you choose `cdecl` ? MSDN seems to say it should be `WINAPI` -> `stdcall`
- https://learn.microsoft.com/en-us/windows/console/handlerroutine
- https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types

Though [this is not relevant](https://cs.opensource.google/go/go/+/master:src/runtime/syscall_windows.go;l=261-263?q=compileCallback&ss=go%2Fgo) for x64 it would still be nice to be consistent.

You can use `windows.NewCallback` for this, You should [prefer](https://pkg.go.dev/syscall) `sys/windows` over `syscall` where possible.